### PR TITLE
Remove old Slider examples and add new ones

### LINKED ISSUE #829: Example: slider
Split appropriate existing examples into separate files using functional approach and icache and move into the correct `/src/examples/widgets` subdirectory.

### PR DESCRIPTION
**Pull request by JamesLMilner**
**Type:** feature

The following has been addressed in the PR:

* [ ] There is a related issue
* [ ] All code matches the [style guide](https://github.com/dojo/framework/blob/master/STYLE.md)
* [ ] Unit tests are included in the PR
* [ ] For new widgets, an entry has been added to the `.dojorc`
* [ ] Any widget variant uses `theme.compose` like [this](https://github.com/dojo/widgets/issues/847)
* [ ] WidgetProperties are exported

**Description:**

This adds examples for the slider widget. It is based off @vansimke commits so props to him! :tada: 

Resolves #829

### CODE REVIEW

File: src/examples/src/widgets/slider/DisabledSlider.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import { icache } from '@dojo/framework/core/middleware/icache';
+import Slider from '@dojo/widgets/slider';
+
+const factory = create({ icache });
+
+export default factory(function DisabledSlider({ middleware: { icache } }) {
+	const value = icache.getOrSet<number>('value', 30);
+	return (
+		<Slider
+			min={0}
```
Comment (tomdye) [inline src/examples/src/widgets/slider/DisabledSlider.tsx]: this isn't disabled

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-qa5hg56qe.now.sh
* Tests Deployment: https://dojowidgets-3ldnuzdo0.now.sh

File: src/examples/src/widgets/slider/DisabledSlider.tsx (hunk 1 above)
Comment (JamesLMilner) [reply src/examples/src/widgets/slider/DisabledSlider.tsx]: :+1:

Comment (JamesLMilner) [reply src/examples/src/widgets/slider/DisabledSlider.tsx]: :+1:

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-locgdwxxv.now.sh
* Tests Deployment: https://dojowidgets-qa5kiyxnc.now.sh

**Pull request merged**